### PR TITLE
[Dev docs] Add bottom app bar docs

### PR DESCRIPTION
**Pull request by guidezpl**

### CODE REVIEW

File: docs/components/bottom_app_bars.md [hunk 1]
```diff
@@ -0,0 +1,113 @@
...
+
+# Bottom app bars
+
+[Bottom app bars](https://material.io/components/app-bars-bottom/) provide access to a bottom navigation drawer and up to four actions, including the floating action button.
+
+![Example bottom app bar with inset floating action button](assets/bottom_app_bars/bottom-app-bar-hero.png)
+
+## Using the bottom app bar
+
+**Note to developers: please add a link to your platform's FAB article below**
```
Comment (perclasson) [inline docs/components/bottom_app_bars.md:18]: I think these notes to developers should be removed.

File: docs/components/bottom_app_bars.md [hunk 2]
```diff
@@ -0,0 +1,113 @@
...
+| **Color** | `color` |
+| **Stroke color** | N/A |
+| **Stroke width** | N/A |
+| **Shape** | `shape` |
+| **Elevation** | `elevation` |
+| **Ripple color** | N/A |
+
+### FAB attributes
+
+**Note to developers: please add a link to your platform's FAB article below**
```
Comment (perclasson) [inline docs/components/bottom_app_bars.md:85]: Remove this.

File: docs/components/bottom_app_bars.md [hunk 3]
```diff
@@ -0,0 +1,113 @@
...
+
+See [floating action button](link to floating action button here).
```
Comment (perclasson) [inline docs/components/bottom_app_bars.md:87]: Link?

File: docs/components/bottom_app_bars.md [hunk 4]
```diff
@@ -0,0 +1,113 @@
...
+For more guidance using the bottom app bar, go to the [Material design page](https://material.io/components/app-bars-bottom/).
+
+## Bottom app bar example
+
+`BottomAppBar`
+* [Class description](https://api.flutter.dev/flutter/material/BottomAppBar-class.html)
+* [GitHub source](https://github.com/flutter/flutter/blob/master/packages/flutter/lib/src/material/bottom_app_bar.dart)
+* [Dartpad demo](https://dartpad.dev/embed-flutter.html?gh_owner=material-components&gh_repo=material-components-flutter&gh_path=docs/components/dartpad/bottom_app_bars/regular)
+
+![Bottom app bar with with leading navigation drawer icon, overlapping FAB with '+' icon, and trailing 'search' and overflow menu control](assets/bottom_app_bars/bottom_app_bar_example.png)
```
Comment (perclasson) [inline docs/components/bottom_app_bars.md:41]: Might want to use the material colors here, like in the bottom navigation: https://github.com/material-components/material-components-flutter/blob/54202c067e9ac07a4b1f20d18b862b3e525a3fc7/docs/components/bottom_navigation.md


 backgroundColor: Color(0xFF6200EE),`

File: docs/components/bottom_app_bars.md (hunk 3 above)
Comment (guidezpl) [reply docs/components/bottom_app_bars.md:87]: I asked in the dev docs chat and they want those to stay there for now

File: docs/components/bottom_app_bars.md [hunk 5]
```diff
@@ -0,0 +1,113 @@
...
+1. Floating action button (FAB)
+1. Action icon
+1. Overflow menu icon
+
+### Container attributes
+
+| &nbsp; | Property |
+| --- | --- |
+| **Color** | `color` |
+| **Stroke color** | N/A |
```
Comment (rami-a) [inline docs/components/bottom_app_bars.md:77]: I think I saw in MH's PR that we should remove N/A attributes

File: docs/components/bottom_app_bars.md [hunk 6]
```diff
@@ -0,0 +1,113 @@
...
+
+
+## Theming
+
+### Theming example
+
+The following example shows a bottom app bar with the [Material Shrine Theme](https://material.io/design/material-studies/shrine.html).
+
+
+* [Dartpad demo](https://dartpad.dev/embed-flutter.html?gh_owner=material-components&gh_repo=material-components-flutter&gh_path=docs/components/dartpad/bottom_app_bars/theme)
```
Comment (rami-a) [inline docs/components/bottom_app_bars.md:108]: In the other docs, we included the code for the themed example in-line too

**Pull request merged**